### ISSUE TITLE
Mismatch in request field names between swagger and gRPC to JSON transcoding

### ISSUE THREAD

**Issue by sartemyev-st** [dotnet/aspnetcore#46670]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

I have gRPC service with message field names in snake_case. After adding json transcoding and swagger schema, it creates swagger.json with camelCase names, but transcoded REST service still expects snake_case names.
It only affects request models. Response models are converted to camelCase in both swagger.json and REST service.

### Expected Behavior

Transcoded REST service should accept requests with camelCase fields, the way they are described in swagger.json.

### Steps To Reproduce

See project which reproduces the issue: [https://github.com/sartemyev-st/grpc-swagger-example](https://github.com/sartemyev-st/grpc-swagger-example)
It contains a simple gRPC service witch responses with the value it receives in request. With json transcoding and swagger.

Open swagger UI and call the service with non empty value in `itemId` parameter.

**Expected result:**
Service responses with the value you send in request

**Actual result:**
Service responses with empty field.

**Notes:**
If you manually open `http://localhost:5045/Service/GetItem?itemId=test` you still receive a response with empty itemId, but if you open `http://localhost:5045/Service/GetItem?item_id=test`, it returns correct value.

### .NET Version

7.0.102

### Anything else?

Package versions:
```xml
        <PackageReference Include="Microsoft.AspNetCore.OpenApi" Version="7.0.3" />
        <PackageReference Include="Swashbuckle.AspNetCore" Version="6.5.0" />
        <PackageReference Include="Microsoft.AspNetCore.Grpc.JsonTranscoding" Version="7.0.3" />
        <PackageReference Include="Microsoft.AspNetCore.Grpc.Swagger" Version="0.3.3" />
        <PackageReference Include="Grpc.Core" Version="2.46.5" />
        <PackageReference Include="Grpc.Tools" Version="2.51.0" />
```

**Comment by athurner** [comment]
We are experiencing the same issue. The problem is primary that gRPC JsonTranscoding does not correctly change field names when used as query parameters. 

I added to the example of @javiercn (see [repo](https://github.com/athurner/grpc-swagger-example)) and demonstrated that in the body the field name is correctly transcoded from snake-case to camel-case - see [the post statement](https://github.com/athurner/grpc-swagger-example/blob/edaf97ddfce12acf912503cc86518abb7f17a3e3/test.http#L11)

But it is not correctly transcoded when used as parameter in the url query.

**Comment by athurner** [comment]
I debugged it with the latest code on the main branch and can confirm that it works now correctly thanks to @JamesNK [changes](https://github.com/dotnet/aspnetcore/commit/c9caab2739ac05843f52b5abe2c105f486ef40a0) 

It would be nice if this could be released in a 7.0.4 release

**Comment by JamesNK** [comment]
It's too late for 7.0.4. Will try to get fixes into 7.0.5.
